library/core/src/slice/iter/macros.rs

```diff
@@ -1,28 +1,55 @@
 //! Macros used by iterators of slice.
 
+macro_rules! zst_len {
+    ($self:ident) => {{
+        #![allow(unused_unsafe)] // we're sometimes used within an unsafe block
+
+        // SAFETY: T is sized, so the raw pointers are always thin, thus the
+        // transmute to `usize` is legal, and we never use the provanance from
+        // the `end` pointer for ZSTs
+        unsafe { mem::transmute::<_, usize>($self.end) }
```

Comment:
Why not `end.addr()`?

Reply:
Oh, right, might as well.  (Means a bit more MIR clutter, but oh well.)

@rustbot author

Reply:
Oh, great, this let me skip the extra macro too since it was only used in a few places, so without the unsafe comment it can just be `.addr()` there since it's in contexts where it's clear enough.

@rustbot ready